### ISSUE TITLE
Updating QueryString on .NET client after connection is started

### ISSUE THREAD

**Issue by cwt237** [SignalR/SignalR#3186]
On the javascript client it is possible to update the QueryString (qs) on the connection object after the connection has been started. 

This is not possible to do with the .NET client, since it only has a getter on the QueryString property.

Now why would I want to do that?

Well, I pass in an OAuth2 token to handle authorization inside the SignalR implementation (using PersistentConnection based implementation). Whenever the token is renewed on the client, the querystring is updated (so the server will see it on the next HTTP request, e.g. on reconnect).

I have made a work around (where the client sends the token directly to the server - and it is then handled and stored as part of the OnReceived handler). However, if the client hits another server - or the server is restarted, the internal storage is gone and the connection cannot be reconnected successfully. 

Long story short: are you planning on exposing a setter on the QueryString property on the .NET client?

In my mind it makes sense to have the same client connection behavior.

**Comment by pkirkley** [comment]
We also have a scenario whereby I would prefer to update the querystring (as part of connect/reconnect events) and prevent having to stop/start (restart) the connection during real-time data messaging. At the moment we are sending a message.  Do you know when this feature would be available please? We are currently running live against 2.1.2.

**Comment by pkirkley** [comment]
Just to confirm I have used the "Headers" as a work around to the above and all is working fine. This is purely to store a simple boolean value to give a true indication if a device client is ready to receive; based on an application consuming our framework that has SignalR comms adapter and then providing the framework with a ready status. We can then provide this status with any of the SignalR events/calls etc

**Comment by Fabsky** [comment]
This feature will be very usefull in case of authentication token that expire.
The suggested workaround (using header) is not possible as websocket doesn't support headers. So currently there is no way to change the querystring once the connection is created ?

**Comment by moozzyk** [comment]
Once you establish a connection using websockets no http requests are being sent except for reconnects or abort so I am not sure where where/why you would change the query string.

**Comment by aspnet-hello** [comment]
This issue has been closed as part of issue clean-up as described in https://blogs.msdn.microsoft.com/webdev/2018/09/17/the-future-of-asp-net-signalr/. If you're still encountering this problem, please feel free to re-open and comment to let us know! We're still interested in hearing from you, the backlog just got a little big and we had to do a bulk clean up to get back on top of things. Thanks for your continued feedback!
